### ISSUE TITLE
Tutorial: ALBERT-large collaborative training

### ISSUE THREAD

**Issue by justheuristic** [learning-at-home/hivemind#219]
Let's implement a basic example for collaborative training with ALBERT
- core training code ( @leshanbog )
  - [x] implement basic training scripts (run_first_peer/run_trainer) based on mryab/collaborative-training 
  - [x] achieve exact match with old training code
  - [x] test fault tolerance against common network failures

- update metric logging code (@yhn112 )
   - [x] tune first peer's wandb to avoid crashes (or restart on crashes)
   - [x] use the same DHT key prefix for metrics and averaging (aka self.prefix)
   - [x] make the prototype pep8-compliant

- add basic security layer: (@borzunov )
   - [x] protect value types in:
      - [x] progress
      - [x] averaging
      - [x] metrics
   - [x] ensure that the averager validate reasonable min/max values (i.e. for batches_processed)
   - [x] make sure it supports DataParallel on peers
   - [x] make sure it will save/load scheduler state dict correctly with optimizer

- add full description in README.md

**Comment by justheuristic** [comment]
Current progress by @leshanbog : https://github.com/learning-at-home/hivemind/tree/collaborative_albert_example

**Comment by yhn112** [comment]
Addition to my part:
Make wandb optional
Better console logging on the coordinator
Log information about individual peers' status to wandb. Needed for debug.

**Comment by justheuristic** [comment]
Addition to addition to your part:
- [ ] modify performance and num peers to count all peers that are currently alive, regardless of step

**Comment by justheuristic** [comment]
It appears that everything in the issue description is done

There are also 3 additions/feature-requests aboveby @yhn112 . Mike, do you consider them solved?

**Comment by yhn112** [comment]
Solved
